main.py

```diff
@@ -74,7 +74,7 @@
         # Initialize DCG
         dcg = dcg_module.DCG(dcg_params)
         # Trains DCG and saves the model
-        dcg_module.train_DCG(dcg, dcg_params, train_loader)
+        dcg_module.train_DCG(dcg, dcg_params, train_loader, test_loader=test_loader)
```

Comment:
Why are you passing test_loader into DCG?

Reply:
To compute eval loss